rfcs/block-component.md

```diff
@@ -52,6 +52,59 @@ Props would be exposed for:
 
 > **Note** that we would likely _not_ use this component internally to build other baseui components. It's primarily a helper component for customers when building their apps.
 
+### `Block` API
+
+* `children: (props: Props) => React.Node` - Optional
+* `as: string | React.Node` - Optional
+  * Modifies the base element used to render the block
+* `$style: Object` - Optional
+  * Applies provided styles to Styletron
+* `color: String` - Optional
+  * Accepts all themeable color properties (`primary200`, etc.)
+* `font: String` - Optional
+  * Accepts all themeable font properties (`font200`, etc.)
```

Comment:
At the expense of adding more to this API, I wonder if we also want to expose props for setting fontSize, fontStyle, fontWeight, lineHeight independently. I suppose people can always do this using `$style`, but I'd imagine it'll be a pretty common use case.

Reply:
I would lean toward allowing people to use `$style` if they need individual overrides.

Since the `fontXXX` theming variables combine all of those values into a single token I think we might want to keep them together for consistency.